src/olympia/blocklist/models.py

```diff
@@ -155,11 +155,34 @@ def get_blocks_from_guids(cls, guids):
         return blocks
 
 
+class BlockVersionQuerySet(models.QuerySet):
+    def _blocked(self, hard):
+        return self.filter(hard=hard, version__file__is_signed=True)
+
+    def soft_blocked(self):
+        return self._blocked(hard=False)
+
+    def hard_blocked(self):
+        return self._blocked(hard=True)
+
+
+class BlockVersionManager(ManagerBase):
+    _queryset_class = BlockVersionQuerySet
+
+    def soft_blocked(self):
+        return self.get_queryset().soft_blocked()
+
+    def hard_blocked(self):
+        return self.get_queryset().hard_blocked()
```

Comment:
You might be able to replace that with just... nothing, and defining `objects` below with [`as_manager()`](https://docs.djangoproject.com/en/4.2/ref/models/querysets/#django.db.models.query.QuerySet.as_manager), like this: `objects = BlockVersionQuerySet.as_manager()`.

For more complex cases where you do want the functionalty `ManagerBase` brings we can also define an empty manager inheriting from `ManagerBase` and use [`from_queryset()`]( (https://docs.djangoproject.com/en/4.2/topics/db/managers/#from-queryset)) like this: `BlockVersionManager.from_queryset(BlockVersionQuerySet)()`

If we have custom methods in the manager that somehow don't belong in the queryset but do call queryset stuff, then it's no longer appropriate, but most of the time that should be enough. We have a bunch of old code using the method you used, but for new cases without anything fancy we should try the modern Django way and avoid repetition.

Reply:
This belongs in the docs.